sway/lock.c

```diff
@@ -0,0 +1,174 @@
+#define _POSIX_C_SOURCE 200809L
+#include "log.h"
+#include "sway/input/keyboard.h"
+#include "sway/input/seat.h"
+#include "sway/lock.h"
+#include "sway/output.h"
+#include "sway/server.h"
+
+#define PERMALOCK_CLIENT (struct wl_client *)(-1)
```

Comment:
Hm, really not a fan of this…

Reply:
I agree, but is there a better way to indicate "sway itself wants to own the lock"?  Adding a bunch of checks if the session is locked in addition to the input-grab is non-null seems worse.

Reply:
Creating a wl_client with a dead or hanging file descriptor would also work, but that seems worse than changing all checks of `exclusive_client` to also check for the lock.